app/packs/src/apps/mydb/inbox/DatasetContainer.js

```diff
@@ -73,104 +73,97 @@ class DatasetContainer extends Component {
+      />
+    ));
+    const attCount = this.attachmentCount();
+
+    const trash = this.props.cache.length === this.props.cache.length // Set it as always show
+      && (
+        <OverlayTrigger
+          show={deletingTooltip}
+          animation
+          trigger="click"
+          placement="bottom"
+          overlay={(<Tooltip placement="bottom" className="in" id="tooltip-bottom">
+            {`Delete ${attCount} attachment${attCount > 1 ? 's' : ''}?`}
+            <ButtonGroup>
+              <Button
+                variant="danger"
+                size="sm"
+                onClick={() => this.confirmDeleteDataset()}
+              >
+                Yes
+              </Button>
+              <Button
+                variant="warning"
+                size="sm"
+                onClick={() => this.toggleTooltip()}
               >
-                <span className="text-info fa fa-arrows ms-1" role="button" />
-              </i>
-              <span style={{ marginLeft: '8px' }}>{dataset.name}</span>
-            </button>
-            {
-              inboxSize && inboxSize !== 'Small'
-              && (
-                <span className="text-info" style={{ float: 'right', display: largerInbox ? '' : 'none' }}>
-                  {formatDate(dataset.created_at)}
-                </span>
-              )
-            }
-          </div>
-          <div>{visible ? attachments : null}</div>
-        </div>,
-        { dropEffect: 'move' }
+                No
+              </Button>
+            </ButtonGroup>
+          </Tooltip>
+          )}
+        >
+          <i
+            className="fa fa-trash-o"
+            onClick={() => this.deleteDataset()}
+            role="button" />
+        </OverlayTrigger>
       );
-    }
-
-    return null;
+    const datasetCheckbox = (
+      <input
+        type="checkbox"
+        checked={isSelected}
+        onChange={() => onDatasetSelect(dataset.id)}
+      />
+    );
+
+    return connectDragSource(
+      <div>
+        <ButtonToolbar className="d-flex align-items-center gap-1 w-100">
+          {datasetCheckbox}
+          {trash}
+          <button
+            type="button"
+            className="border-0 bg-transparent"
+            onClick={() => this.setState(prevState => ({ ...prevState, visible: !visible }))}
+          >
+            <i
+              className={`fa fa-folder${visible ? '-open' : ''}`}
+              role="button"
+            >
+              <span className="text-info fa fa-arrows ms-1" role="button" />
+            </i>
+            <span className="ms-1">{dataset.name}</span>
+          </button>
+
+          {
+            inboxSize && inboxSize !== 'Small'
+            && (
+              <span className={`text-info ms-auto ${largerInbox ? '' : 'none'}`}>
+                {formatDate(dataset.created_at)}
+              </span>
+            )
+          }
+        </ButtonToolbar>
+        <div>{visible ? attachments : null}</div>
```

Comment:
Can this be converted to `{visible ?? <div>…}` so we don't render the empty `<div>`?